L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -859,10 +844,10 @@ void HGCalTriggerGeometryV9Imp3::fillMaps() {
         lpgbt_to_modules_.emplace(lpgbt_id, packed_value);
 
         //fill subsiduary module to stage 1 mapping
-        if (module_to_stage1_.find(packed_value) == module_to_stage1_.end()) {
-          module_to_stage1_.emplace(packed_value, stage1_id);
-        } else {  //check that the stage1_id is the same as in the existing map
-          if (stage1_id != module_to_stage1_.at(packed_value)) {
+        auto result = module_to_stage1_.emplace(packed_value, stage1_id);
+        if (result.second == false) {  //check that the stage1_id is the same as in the existing map
+          //	  auto &ref_to_ddd = result.first->second;
+          if (stage1_id != result.first->second) {
```

Comment:
This could be done in the first `if` statement:
```cpp
if(result.second == false && stage1_id != result.first->second)
```

Reply:
done